stat/stat.go

```diff
@@ -1422,6 +1426,50 @@ func linInterpQuantile(p float64, x, weights []float64, sumWeights float64) floa
 	panic("impossible")
 }
 
+func linearQuantile(p float64, x, weights []float64, sumWeights float64) float64 {
+	n := len(x)
+
+	// Unweighted: Hyndman & Fan Type 7 (R default).
+	if weights == nil {
+		if p <= 0 {
+			return x[0]
+		}
+		if p >= 1 {
+			return x[n-1]
+		}
+		h, frac := math.Modf(1 + p*float64(n-1))
+		i := int(h) - 1
+		if frac == 0 {
+			return x[i]
+		}
+		return (1-frac)*x[i] + frac*x[i+1]
```

Comment:
```suggestion
		return float64((1-frac)*x[i]) + float64(frac*x[i+1])
```

Reply:
You could probably make an argument that we could make use of:

``go
math.FMA(frac, float64(x[i+1]-x[i]), float64(x[i]))
```
 
Which was added in Go 1.14 and is more idiomatic ...